src/binary/timestamp.rs

```diff
@@ -18,7 +18,7 @@ use crate::{
 impl Timestamp {
     const BUFFER_SIZE: usize = 16;
 
-    /// NOTE: Currently, this function always encodes with nanosecond precision.
+    /// NOTE: If the Timestamp being encoded has fractional seconds, they will be written as nanoseconds.
```

Comment:
So this means that if there exists sub-second precision, that precision is always nine digits (i.e. nanoseconds)?

Reply:
Yeah, for now.

My overall philosophy right now is I'm trying to extract/de-duplicate code between ion-rust and ion-hash. (So, the nanosecond precision caveat was already there.) Happy to open and fix these issues later, but given how little time I have to work on this (and likewise, your review time) I'm trying to take babysteps.

Reply:
No worries--just making sure I understood the comment.

Reply:
Added #274.